### DISCUSSION TITLE
Houseimint

### DISCUSSION THREAD

**Discussion by humansinstitute** [fedimint/fedimint#2549] [Fedimint Modules]
We often talk about the benefits of making small micro payments in Fedimints, but no large payments.

Recently I had to make reasonably large payments both in a Fedimint and over the main chain and it struck me there was a real use case for solving large value transfers in a time critical manner through Fedimints. 

The main use case I had in mind was house purchases.

One of the most frustrating parts of purchasing a house is the settlement and key exchange. Here you have to arrange a day for the transaction, prep in advance to have all you money in place from the mortgage provider before relying on the will of two banks to hopefully make an exchange "at some point on a Friday otherwise you won't have a house for the weekend."

Whilst high fees are annoying, the real worry here is the timing is so opaque and difficult and I believe this would be a good use case for a Fedimint with speciality Guardians e.g. four to seven trusted institutions who would simply run this as an escrow settlement service at a much lower cost to competitors. 

The benefit here is you could make the transaction instant, limit the amount of time you are exposed to the counter party risk of the housimint, you avoid block confirmation of main chain and you avoid routing failures for high value lightning payments.

You could even use additional modules to make the process simpler or safer.

### How might it work?

1. You and your counter party agree on, one of many housimints in your area. 
2. Prior to the agreed settlement data both you deposit bitcoin to Mint (either one on-chain or many lightning payments streamed in). 
3. Exchange of funds is made and legal settlement complete
4. The counter party can choose to either withdraw via on chain fee or stream out via lightning at their discretion. 

### Could it work with other modules?

Yep you could use stability pools or stable assets to limit the exposure to price volatility. You could use an automated escrow contract to manage the transfer. You could 

#### FAQ

**Why not use just use lightning?**
Good question. Lightning can route a million sats, which should buy a house or two in 10 years ;) - but it is subject to routing issues with higher payments. 

The idea here is that you can abstract the actual point of settlement away from the payment and user taking custody of their money either on chain, via lightning or back to their local fedimint. 

**Why not use on chain?**
Is it really better to sit for an hour waiting for confirmations instead of just letting two banks do it, when you could temporarily take advantage of a trusted business to facilitate this for pennies with minimal exposure (maybe you spend an extra day or two in the housimint as your funds accumulate over lightning) and low fees.

Also don't be bearish, bitcoin fees will be higher than buying a house :)

**Comment by jkitman** [comment]
Not sure why we need instant house settlement considering the legal paperwork takes days generally.  Besides if you believe fiat will debase more than the interest rate over your loan duration you are probably better off financing :smiley: 

But to solve your original problems, it's probably safer for all parties to use Fedimint as an escrow service in a 2-of-3 multisig where the Fedimint acts as one of the cosigners (e.g. #1455).  Otherwise, the rug potential is pretty high.
